<launch>
  <arg name="param_file_path" default="$(find gnss_area_recognizer)/config/gnss_area_recognizer.yaml" />

  <arg name="input_lanelet2_map_bin_topic" default="/map/vector_map" />
  <arg name="input_pose_with_covariance_topic" default="/localization/pose_twist_fusion_filter/pose_with_covariance" />

  <arg name="output_pose_with_covariance_topic" default="/localization/util/gnss_area_recognizer/pose_with_covariance" />

  <node pkg="gnss_area_recognizer" type="gnss_area_recognizer" name="gnss_area_recognizer">
    <remap from="input_lanelet2_map_bin_topic" to="$(arg input_lanelet2_map_bin_topic)" />
    <remap from="input_pose_with_covariance_topic" to="$(arg input_pose_with_covariance_topic)" />

    <remap from="output_pose_with_covariance_topic" to="$(arg output_pose_with_covariance_topic)" />

    <rosparam command="load" file="$(arg param_file_path)" />
  </node>
</launch>
